### PR TITLE
Added unit tests for SampleEnvSetter.

### PR DESCRIPTION
**Pull request by harshitsharma2801**
1. Added Junit5 to POM.XML
2. Added maven-surefire-plugin to run tests on every build
3. POM.XML was formatted by a XML Formatter
4. Added a setter to set mqEndpoints to mock and test the endpoints from SampleEnvSetterTest.

### CODE REVIEW

File: JMS/src/test/java/com/ibm/mq/samples/jms/SampleEnvSetterTest.java [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+import org.json.JSONArray;
+import org.json.JSONObject;
+import java.util.ArrayList;
+import java.util.List;
+
+public class SampleEnvSetterTest {
+
+    static SampleEnvSetter envSetter;
+    @BeforeAll
+    public static void setUp(){
```
Comment (chughts) [inline JMS/src/test/java/com/ibm/mq/samples/jms/SampleEnvSetterTest.java:32]: Our code makes use of an `env.json` file overridden by environment variables. The tests should take that into account. ie. fails if file can't be found. 

The code allows overriding of the env file location and name by
`String valueEnvFile = System.getProperty(ENV_FILE);` the tests should be making sure that works.

**Review by chughts — CHANGES_REQUESTED** [review]

**Comment by harshitsharma2801** [comment]
1. Removed the setter function
2. Updated Readme
3. Added test for Env file

File: JMS/com/ibm/mq/samples/jms/SampleEnvSetter.java [hunk 2]
```diff
@@ -1,5 +1,5 @@
 /*
-* (c) Copyright IBM Corporation 2019, 2023
+* (c) Copyright IBM Corporation 2019, 2024
```
Comment (chughts) [inline JMS/com/ibm/mq/samples/jms/SampleEnvSetter.java:2]: This file is no-longer changed, revert the copyright date.

File: JMS/src/test/java/com/ibm/mq/samples/jms/SampleEnvSetterTest.java [hunk 3]
```diff
@@ -0,0 +1,201 @@
...
+import org.json.JSONObject;
+import java.util.ArrayList;
+import java.util.List;
+
+public class SampleEnvSetterTest {
+
+    static SampleEnvSetter envSetter;
+    
+    @BeforeAll
+    public static void setUp() throws NoSuchFieldException, IllegalAccessException{
```
Comment (chughts) [inline JMS/src/test/java/com/ibm/mq/samples/jms/SampleEnvSetterTest.java:34]: I don't think we should be mocking the endpoints. The point of the test is to verify:
1. That the env setter is able to read the file, whose name and location are specified by an environment variable. 
2. That it allows overrides via environment variables.
3. How it behaves when there are missing values in the env file.
4. How it behaves when the file can't be found.

**Review by chughts — CHANGES_REQUESTED** [review]

File: JMS/src/test/java/com/ibm/mq/samples/jms/SampleEnvSetterTest.java [hunk 4]
```diff
@@ -17,139 +17,67 @@
...
-import java.lang.reflect.Field;
-
 import org.junit.jupiter.api.Test;
 import org.junit.jupiter.api.BeforeAll;
-import org.json.JSONArray;
-import org.json.JSONObject;
-import java.util.ArrayList;
-import java.util.List;
 
 public class SampleEnvSetterTest {
```
Comment (chughts) [inline JMS/src/test/java/com/ibm/mq/samples/jms/SampleEnvSetterTest.java:23]: These are private strings in `SampleEnvSetter`. 

```
    private static final String ENV_FILE = "EnvFile";
    private static final String DEFAULT_ENV_FILE = "../env.json";    
    private static final String DEFAULT_Z_ENV_FILE ="../env-zbindings.json";
``` 
In this case, there is an argument to make them public. So the code in the test makes use of them.

File: JMS/src/test/java/com/ibm/mq/samples/jms/SampleEnvSetterTest.java [hunk 5]
```diff
@@ -17,139 +17,67 @@
...
     @Test 
     public void testGetPortEnvValue(){
-        //Test for correct port key
+        //Test for port key
         int value = envSetter.getPortEnvValue("PORT", 0);
-        assertEquals(1415 , value);
+        assertNotNull(value);
 
         //Test for default port given invalid port key
         value = envSetter.getPortEnvValue("INVALID_PORT", 0);
```
Comment (chughts) [inline JMS/src/test/java/com/ibm/mq/samples/jms/SampleEnvSetterTest.java:78]: why is `INVALID_PORT` returning 1414 ?

**Review by chughts — CHANGES_REQUESTED** [review]

File: JMS/src/test/java/com/ibm/mq/samples/jms/SampleEnvSetterTest.java (hunk 5 above)
Comment (harshitsharma2801) [reply JMS/src/test/java/com/ibm/mq/samples/jms/SampleEnvSetterTest.java:78]: The function returns the default port if the key is not found.

File: JMS/src/test/java/com/ibm/mq/samples/jms/SampleEnvSetterTest.java [hunk 6]
```diff
@@ -22,13 +22,13 @@
 
 public class SampleEnvSetterTest {
 
-    static SampleEnvSetter envSetter;
+    private static SampleEnvSetter envSetter;
     private static final String ENV_FILE = "EnvFile";
```
Comment (chughts) [inline JMS/src/test/java/com/ibm/mq/samples/jms/SampleEnvSetterTest.java:26]: These are no longer use, they can be removed.

**Review by chughts — CHANGES_REQUESTED** [review]

**Review by chughts — APPROVED** [review]

**Review by rcoppen — APPROVED** [review]

**Pull request merged**